services/backend/src/core/profile/profile.js

```diff
@@ -879,6 +884,26 @@ async function getFullProfile(id, language) {
       },
     },
   });
+
+  fullProfile.secondLangProfs.forEach((value, index) => {
+    let expiredValue;
+    if (value.date) {
+      const dateMoment = moment(value.date);
+      if (dateMoment.isBefore()) {
+        expiredValue = true;
+        if (dateMoment.unix() === 0) {
+          fullProfile.secondLangProfs[index].date = null;
+        }
+      } else {
+        expiredValue = false;
+      }
+    } else {
+      expiredValue = null;
+    }
+
+    fullProfile.secondLangProfs[index].expired = expiredValue;
+  });
```

Comment:
I would put this in the `filterProfileResult` function, this function should be kept to fetch the data from the db

Reply:
moved to filter